components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -871,6 +881,7 @@ public IdentityProvider getIdPByResourceId(String resourceId, String tenantDomai
         validateGetIdPInputValues(resourceId);
         int tenantId = IdentityTenantUtil.getTenantId(tenantDomain);
         IdentityProvider identityProvider = dao.getIdPByResourceId(resourceId, tenantId, tenantDomain);
+        populateEndpointConfig(identityProvider, tenantDomain);
```

Comment:
IMO, we can extract these two to one single method. Or wrap the dao from a layer which returns the proper IdP model